src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -19,87 +24,242 @@
 
 JGI_GENOME_LOOKUP_URL = 'https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid={}'
 
 USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0) Gecko/20100101 Firefox/86.0'
 
+GENOME_STATUS_FILENAME = "genome_status.csv"
 
-class GenomeStatus:
 
-    def __init__(self, original_id, resolved_id, attempted=False, filename=""):
+class GenomeStatus:
+    """To represent the data status for a certain genome ID.
+
+    Attributes:
+        original_id(str): original ID for the genome. 
+        resolved_refseq_id(str): RefSeq ID generated by resolving the original genome ID. 
+        attempted(str): If the attempt of resolving the genome ID has already been done. Defaults to `False`. 
+        filename(str): Path to the file that stores the genome data. Defaults to "". 
+    """
+
+    def __init__(self,
+                 original_id: str,
+                 resolved_refseq_id: str,
+                 attempted: str = 'False',
+                 filename: str = ""):
         self.original_id = ';'.join(original_id.split(','))
-        self.resolved_id = None if resolved_id == 'None' else resolved_id
-        self.attempted = True if attempted == 'True' else False
+        self.resolved_refseq_id = "" if resolved_refseq_id == 'None' else resolved_refseq_id
+        self.attempted = attempted == 'True'
         self.filename = filename
 
-    @classmethod
-    def from_csv(cls, original_id, resolved_id, attempted, filename):
-        return cls(original_id, resolved_id, attempted, filename)
-
-    def to_csv(self):
-        return ','.join([
-            str(self.original_id),
-            str(self.resolved_id),
+    def to_csv(self, file: str | PathLike, mode: str = 'a') -> None:
+        """Write object to a csv file.
+        """
+        line = ','.join([
+            self.original_id, self.resolved_refseq_id,
             str(self.attempted), self.filename
         ])
+        with open(file, mode) as f:
+            f.write(line + '\n')
+
+
+def podp_download_and_extract_antismash_data(
+        genome_records: list[dict[str, dict[str, str] | str]],
+        project_download_root: str | Path, project_extract_root: str | Path):
+    """Download and extract antiSMASH BGC archive for the genomes specified in genome_records
+    for which the original genome ID can be resolved.
+
+    Args:
+        genome_records(list[dict[str, dict[str, str]  |  str]]): list of dicts representing genomes'
+        records. Each record contains information about the specific genome's ID. 
+        project_download_root(str | Path): Path to the directory to place downloaded archive in.
+        project_extract_root(str | Path): Path to the directory data files will be extracted to.
+            Note that an `antismash` directory will be created in the specified `extract_root` if
+            it doesn't exist. The files will be extracted to `<extract_root>/antismash/<antismash_id>` directory.
+    """
```

Comment:
```suggestion
    """Download and extract antiSMASH BGC archive for the given genome records.
	
    Args:
        genome_records(list[dict[str, dict[str, str] | str]]): list of dicts 
			representing genome records. The dict of each genome record contains
				- key: [PLEASE add details]
				- value: [PLEASE add details]
        project_download_root(str | Path): Path to the directory to place 
			downloaded archive in.
        project_extract_root(str | Path): Path to the directory downloaded archive 
			will be extracted to.
            Note that an `antismash` directory will be created in the specified 
			`extract_root` if it doesn't exist. The files will be extracted to
			`<extract_root>/antismash/<antismash_id>` directory.
    """
```

The summary of docstring should be only one line. More details can be provided in a new paragraph after it or in the arguments.
Please fill the details for the arg `genome_records`.